client/src/routes.tsx

```diff
@@ -1,204 +1,299 @@
-import TrackGroup from "components/TrackGroup/TrackGroup";
-import About from "components/pages/About";
 import { AuthWrapper } from "components/AuthWrapper";
-import Signup from "components/Signup";
-import ManageTrackGroup from "components/ManageArtist/ManageTrackGroup";
-import Releases from "components/Releases";
-import ManageContainer from "components/ManageArtist/ManageContainer";
-import ManageArtistContainer from "components/ManageArtist/ManageArtistContainer";
-import ArtistContainer from "components/Artist/ArtistContainer";
-import ProfileContainer from "components/Profile/ProfileContainer";
-import WishlistCollection from "components/Profile/WishlistCollection";
-import MarkdownContent from "components/common/MarkdownContent";
-import Welcome from "components/ManageArtist/Welcome";
-import ArtistPosts from "components/Artist/ArtistPosts";
-import ArtistAlbums from "components/Artist/ArtistAlbums";
-import ArtistSupport from "components/Artist/ArtistSupport";
-import ManageArtistAlbums from "components/ManageArtist/ManageArtistAlbums";
-import ManageArtistSubscriptionTiers from "components/ManageArtist/ManageArtistSubscriptionTiers";
-import ManageArtistPosts from "components/ManageArtist/ManageArtistPosts";
-import ManagePost from "components/ManageArtist/ManagePost";
-import NewReleaseRedirect from "components/ManageArtist/NewReleaseRedirect";
 import { css } from "@emotion/css";
-import Supporters from "components/ManageArtist/Supporters";
-import Artists from "components/Artists";
-import AdminArtists from "components/Admin/AdminArtists";
-import ManageArtistAlbumTools from "components/ManageArtist/ManageArtistAlbumTools";
-import Features from "components/Home/Features";
-import RedeemCode from "components/TrackGroup/RedeemCode";
-import DownloadAlbum from "components/TrackGroup/DownloadAlbum";
-import AdminPurchases from "components/Admin/AdminPurchases";
-import AdminSubscriptions from "components/Admin/AdminSubscriptions";
-import CallServerTasks from "components/Admin/CallServerTasks";
-import AdminSettings from "components/Admin/AdminSettings";
-import UserNotificationFeed from "components/Profile/UserNotificationFeed";
-import AdminManageUser from "components/Admin/AdminManageUser";
-import ArtistUnsubscribe from "components/Artist/ArtistUnsubscribe";
 
-const routes = [
+async function markdownPage(source: string) {
+  const { PageMarkdownWrapper } = await import("components/Post");
+  const { default: MarkdownContent } = await import(
+    "components/common/MarkdownContent"
+  );
+  return {
+    Component: () => (
+      <PageMarkdownWrapper>
+        <MarkdownContent source={source} />
+      </PageMarkdownWrapper>
+    ),
+  };
+}
+
+const routes: RouteObject[] = [
   {
     path: "/",
     element: <App />,
     errorElement: <ErrorPage />,
     children: [
-      { path: "", element: <Home /> },
-      { path: "pages/about", element: <About /> },
+      {
+        path: "",
+        async lazy() {
```

Comment:
Do you know how this will play with pre-renderers?

Reply:
Nope! Does Mirlo currently use SSR? (I didn't think it did, but could be wrong)

[According to the react-router docs](https://reactrouter.com/en/main/guides/ssr#lazy-routes) it seems like lazy functions are evaluated for SSR, but might cause some issues when hydrating on the client. This could be fixed by manually awaiting the correct route functions on initial load (as shown in their code sample) - or by using the full [Remix](https://remix.run) framework, which is their suggestion.

I'm not sure how much would be needed to migrate to Remix - from their docs it looks like it's just adding a Vite plugin and then moving the routes into file-based route modules, but there could be other differences.

Reply:
> Nope! Does Mirlo currently use SSR? (I didn't think it did, but could be wrong)

We don't, but we use netlify's [prerendering](https://docs.netlify.com/site-deploys/post-processing/prerendering/) for places that don't load the full page -- ie social media, discord, etc

It's fairly easy to test though, we can use the netlify preview to test it in discord